fastestimator/trace/io/model_eval.py

```diff
@@ -0,0 +1,446 @@
+        report_name = re.sub('_{2,}', '_', report_name) + "_test_report"
+        self.report_name = report_name or 'test_report'
+
+    def _init_document(self) -> None:
+        """Initialize latex document
+        """
+        self.doc = Document(geometry_options=['lmargin=2cm', 'rmargin=2cm', 'tmargin=2cm', 'bmargin=2cm'])
+        self.doc.packages.append(Package(name='placeins', options=['section']))
+        self.doc.packages.append(Package(name='float'))
+        self.doc.preamble.append(NoEscape(r'\usetikzlibrary{positioning}'))
+
+        self.doc.preamble.append(NoEscape(r'\aboverulesep=0ex'))
+        self.doc.preamble.append(NoEscape(r'\belowrulesep=0ex'))
+        self.doc.preamble.append(NoEscape(r'\renewcommand{\arraystretch}{1.2}'))
+
+        self.doc.preamble.append(Command('title', self.json_summary["title"]))
+        self.doc.preamble.append(Command('author', f"FastEstimator {fe.__version__}"))
+        self.doc.preamble.append(Command('date', NoEscape(r'\today')))
+        self.doc.append(NoEscape(r'\maketitle'))
+
+        # new column type
+        self.doc.preamble.append(NoEscape(r'\newcolumntype{Y}{>{\centering\arraybackslash}X}'))
+
+        # add tabularx hyphentation
+        self.doc.preamble.append(NoEscape(r'\def\seqinsert{\-}'))
+
+        # TOC
+        self.doc.append(NoEscape(r'\tableofcontents'))
+        self.doc.append(NoEscape(r'\newpage'))
+
+    def _document_test_result(self) -> None:
+        """Document test results including test summary, passed tests, and failed tests.
+        """
+        self.test_id = 1
+        instance_pass_tests, aggregate_pass_tests, instance_fail_tests, aggregate_fail_tests = [], [], [], []
+
+        for test in self.json_summary["tests"]:
+            if test["test_type"] == "per-instance" and test["passed"] == True:
+                instance_pass_tests.append(test)
+            elif test["test_type"] == "per-instance" and test["passed"] == False:
+                instance_fail_tests.append(test)
+            elif test["test_type"] == "aggregate" and test["passed"] == True:
+                aggregate_pass_tests.append(test)
+            elif test["test_type"] == "aggregate" and test["passed"] == False:
+                aggregate_fail_tests.append(test)
+
+        with self.doc.create(Section("Test Summary")):
+            with self.doc.create(Itemize()) as itemize:
+                itemize.add_item(
+                    escape_latex("Execution time: {:.2f} seconds".format(self.json_summary['execution_time(s)'])))
+
+            with self.doc.create(Table(position='H')) as table:
+                table.append(NoEscape(r'\refstepcounter{table}'))
+                self._document_summary_table(pass_num=len(instance_pass_tests) + len(aggregate_pass_tests),
+                                                fail_num=len(instance_fail_tests) + len(aggregate_fail_tests))
+
+        if len(instance_pass_tests) + len(aggregate_pass_tests) > 0:
```

Comment:
could just do 

if instance_pass_tests or aggregate_pass_tests:

no need for len()

Reply:
Ya it is more elegant.

Reply:
fixed